cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/spec/PipelineSpecGenerator.java

```diff
@@ -127,6 +148,15 @@ protected void configureStages(ETLConfig config, PipelineSpec.Builder specBuilde
       .setNumOfRecordsPreview(config.getNumOfRecordsPreview());
   }
 
+  private boolean hasSameSchema(Map<String, Schema> inputSchemas, Schema inputSchema) {
+    if (!inputSchemas.isEmpty()) {
+      if (!Objects.equals(inputSchemas.values().iterator().next(), inputSchema)) {
```

Comment:
are all schemas in the iterator guaranteed to be the same?
If not, then why are we only checking the first schema?